src/piece_picker.cpp

```diff
@@ -2117,7 +2122,7 @@ namespace {
 				}
 				else
 				{
-					for (piece_index_t i = m_cursor; i < m_reverse_cursor; ++i)
+					for (piece_index_t i = m_cursor > sequential_start ? m_cursor : sequential_start; i < m_reverse_cursor; ++i)
```

Comment:
Wouldn't it be better to set `m_cursor` instead of having this check every time?

Reply:
some consideration should be given the case where sequential mode is enabled, and `m_cursror == m_reverse_cursor` but not all pieces have been downloaded. It seems suspicious if the sequential start would implicitly set some pieces to "do not download" priority.

Reply:
> Wouldn't it be better to set m_cursor instead of having this check every time?

I thought about it.
It works for me like this. Since the starting piece can change at any time. It would be best to check this every time. This will increase the dynamics of loading the necessary data.

I leave it up to you to choose how to do this.

> some consideration should be given the case where sequential mode is enabled, and m_cursror == m_reverse_cursor but not all pieces have been downloaded. It seems suspicious if the sequential start would implicitly set some pieces to "do not download" priority.

Maybe that's really the problem. What do you suggest as a solution?

Reply:
> Maybe that's really the problem. What do you suggest as a solution?

I think disabling sequential mode and resetting the boundaries would be appropriate.

Reply:
@arvidn Okay. This can be done by other methods. There is no need to do this inside the library.